File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    # x는 가로 y는 세로 가로가 더 크거나 같음
    for x in range(3,brown//2):
        for y in range(3,brown//2):
            if(((2*x)+(2*y)-4==brown) and ((x-2)*(y-2)==yellow) and(x>=y)):
                return [x,y]

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        // 전체 격자 수 = brown + yellow
        int total = brown + yellow;
        
        // yellow의 약수 쌍 (w, h) 탐색
        for (int h = 1; h <= Math.sqrt(yellow); h++) {
            if (yellow % h == 0) {
                int w = yellow / h; // w * h = yellow
                // 갈색 격자 수 확인: 2w + 2h + 4 == brown
                if (2 * w + 2 * h + 4 == brown) {
                    // 가로 >= 세로, 전체 가로 = w + 2, 세로 = h + 2
                    return new int[]{w + 2, h + 2};
                }
            }
        }
        
        // 제한사항 상 답이 항상 존재
        return new int[]{};
    }
}